### DISCUSSION TITLE
Cannot damage same-sub external walls while inside with standard guns

### DISCUSSION THREAD

**Discussion by Dunkle-Stiltskin** [FakeFishGames/Barotrauma#14422] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

While within the bounds of a submarine, some weapons cannot damage Wall pieces that are outside of the same submarine; mostly limited to walls not connected to any hulls, but even "shell" Walls  _can_ be affected.

"Inside" the submarine, for the _player_, includes: inside a hull in dry air , inside a flooded hull, and swimming in the thin layer of water outside the hull (where your movement follows the submarine and are pulled with it) though this area has inconsistent behavior.

"Outside" the submarine is "open ocean", (where you are NOT dragged with submarine)

Shell walls are walls at outer edge of submarine that are partly within a hull, and partly in open water, these are the walls that if damaged or broken result in a Hull Breach and flooding.
These shell walls are _inconsistent_ when testing this bug, and _some walls_ won't be damaged while still inside the submarine; sometimes even the _same_ wall changes behavior depending on exact shot angle. (Yellow in overview image).

This affects the many common "void areas" consisting of only background without Hull that many submarines have that separate the interior walls from the outer shell.  **This bug reliably results in any such layered walls resulting in it being sometimes _impossible_ to break through all the walls from the inside** - which may be desired if trying to escape. 

Any parts of a _different_ submarine than the one you are inside are damaged without issue.
Attacking while in "open ocean" outside the submarine deals damage without issue.

Issue occurs from any direction of attack, reversing the test and attacking exterior walls from above has identical effects.

Issue occurs with all submarines tested, so is not specific to a particular submarine.

Weapons tested where issue WAS observed (did not damage exterior walls while inside):
-Assault Rifle (standard ammo, demonstrated in attached videos)
-SMG (standard ammo, depleted ammo)
-Revolver (standard ammo, depleted ammo)
-Handcannon
-Riot & Autoshotgun (Shotgun Shells)
-HMG
-Rapid Fissile Accelerator ( **very** surprisingly , no damage at all to affected walls)

Weapons tested where issue was NOT observed (behaved as expected): 
-Boarding Axe
-Crowbar
-Plasma Cutter
-Harpoon Gun (Harpoon, Physicorium Harpoon)
-Riot & Autoshotgun (Explosive Slugs (explosive component DOES cause damage, impact component does not?)
-Grenade Launcher (40mm grenades) (explosive & impact damage DOES damage, no issue)
-Turrets placed inside submarine, with player manning periscope inside submarine.

![Overview image](https://github.com/user-attachments/assets/4c00fffd-914b-4a32-ad11-48a9a665e90b)

When making layered walls for any reason, you don't want multiple layers of wall to overlap the same edge of a Hull, because if both walls are broken then the flooding that is caused is _doubled_ in speed (1x for each wall broken) for the same length of broken wall.  So to make layered walls you must place the outer layer such that it does not overlap the Hull edge; this triggers a situation the bug can arise.

Example of simple case where bug may appear:
Below window is 2-layered: a thick layer at bottom that is connected to hull, only it being damaged causes breach; and a thinner top layer that protects the bottom layer from initial damage.  Standing below this window and shooting up, you will destroy the bottom window (flooding will occur), but continuing to shoot will NOT be able to damage the top window.
![image](https://github.com/user-attachments/assets/16db6960-bcc5-4bd2-a3b2-f98af16cf9ed)

Demonstration while in engine room of Humpback:
https://github.com/user-attachments/assets/d6592a55-e0bf-40d4-967f-16993b413450

Demonstration while submerged in water while within the submarine (by destroying hull shell):
https://github.com/user-attachments/assets/584a29f4-9032-43ca-9e04-b7d0a8c83fc0

Demonstration while in dry air while within the submarine (by having no gap or wall between outside):
https://github.com/user-attachments/assets/4bad11ee-5249-443f-92ce-fc1c720ca878

### Reproduction steps
To see on an existing submarine:
In the vanilla Humpback, from anywhere in the engine room, attack through the ceiling wall then at the shell above you.
![image](https://github.com/user-attachments/assets/3501335a-854e-4e38-b79e-9dc2e6949b13)

In the vanilla Typhoon are a couple of affected locations, stand anywhere within the Brig or hallway next to it, and shoot through the wall then at the shell underneath you.  
![image](https://github.com/user-attachments/assets/dfc91ad8-0062-4686-8dab-54cb67bfaa15)

You will see that the inside wall can be attacked, but the shell cannot be destroyed without exiting the Hull area.  Even within the dark background in-between, it is _very_ difficult to land hits on the outer shell that deal damage (with affected weapons).

For custom demonstration:
Using submarine editor, with any conventional submarine; find a flat ceiling (any direction wall will work) to use as a testing zone.  
Place wall pieces (glass prefered for easy viewing) outside of the submarine between a couple meters (at least not intersecting a hull edge) to a few dozen meters away (~within view range); these pieces should have no hull connected to them.

Activate simulation, and grab a suit (or pressure stabilizer) and any weapon from above problem list.

Stand in the test zone, and begin shooting through the hull wall until it is fully breached, 
then try to shoot the outside wall that you placed with the same weapon and observe that _it just doesn't work._

Below heavily-stripped submarine that I used for testing (and the demo) has situation already setup; including the proper example window previously mentioned.

[Berilia  - BUG DEMO - stripped.zip](https://github.com/user-attachments/files/16568808/Berilia.-.BUG.DEMO.-.stripped.zip)

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
No console errors appear that I can find, nor crash reports/logs.
```

**Comment by Ek-Videogames** [comment]
I have also noticed this problem on Unstable but the opposite, exterior guns not causing damage to interior walls:

On some vanilla submarines (ones where the firing arc of a coilgun allows the submarine to shoot its own hull) the projectiles will collide with an interior wall or inner-airlock door and disappear without doing any damage.

**Comment by gerakis** [comment]
I have a similar problem where if I dock with a beacon (modded in) I am unable to plasma cut its doors open from inside my own submarine. If I undock and move away a bit, then swim out I can easily cut through the door. I suspect its the same bug.

**Comment by Regalis11** [comment]
The issue has been addressed in our private repo now, fix coming up in the next unstable build and the upcoming update! :)
